src/icmp/Icmp4.cc

```diff
@@ -132,6 +132,11 @@ Icmp4::SendEcho(Ip::Address &to, int opcode, const char *payload, int len)
     icmp->icmp_cksum = CheckSum((unsigned short *) icmp, icmp_pktsize);
 
     to.getAddrInfo(S);
+    if (!S || !S->ai_addr || S->ai_family != AF_INET) {
```

Comment:
AFAICT, we should `Assure(S)` rather than imply that it may not be set. getAddrInfo() promises to allocate that structure (and does).

Reply:
sounds reasonable. I've updated it here and in icmp6